libnmstate/nm/connection.py

```diff
@@ -294,6 +300,28 @@ def waitfor_active_connection_async(self, ac):
                 "state-changed", self._waitfor_device_state_change_callback, ac
             )
         )
+        # As we also depend on IP4_READY flag which does not trigger
+        # 'state-change' signal, hence do a repeated recheck without timeout.
+        ac.state_check_running = True
+        nmclient.GLib.timeout_add_seconds(
+            ConnectionProfile._ACTIVATION_CHECK_INTERNAL,
+            self._check_device_state,
+            ac,
+        )
+        logging.debug("Activation state check started")
+
+    def _check_device_state(self, ac):
+        if ac.state_check_running and (
+            self._activation_check_count
+            < ConnectionProfile._MAX_ACTIVATION_CHECK_COUNT
+        ):
+            self._activation_check_count += 1
```

Comment:
Can we move this counter through the callbacks? It looks ugly to do it on the object state.